### PR TITLE
Updates for owner delete

### PR DESCRIPTION
**Pull request by wottop**
If share org is deleted, then the share consumer is removed from master org
If REST call comes to delete master org while there is a share, then it will
 fail with a message unless the force query parameter is used.

### CODE REVIEW

**Comment by awood** [comment]
bot, retest this please

File: server/src/main/java/org/candlepin/model/ConsumerCurator.java [hunk 1]
```diff
@@ -338,6 +338,25 @@ public Consumer getConsumer(String uuid) {
...
+
+        return this.cpQueryFactory.<Consumer>buildQuery(this.currentSession(), criteria);
+    }
+
+    public boolean doesShareConsumerExist(Owner owner) {
+        long result = (Long) createSecureCriteria()
+            .add(Restrictions.eq("owner", owner))
+            .add(Restrictions.isNotNull("recipientOwnerKey"))
+            .setProjection(Projections.count("id"))
+            .uniqueResult();
```
Comment (awood) [inline server/src/main/java/org/candlepin/model/ConsumerCurator.java]: `uniqueResult` returns null if there is not a result, not 0.  I would just drop `uniqueResult` entirely, grab the count, and `return count != 0`

Comment (awood) [inline server/src/main/java/org/candlepin/model/ConsumerCurator.java]: This is an invalid comment since `count` will always return something.

**Comment by wottop** [comment]
jenkins: retest this please

**Comment by awood** [comment]
Bot, retest this please

**Comment by wottop** [comment]
[Multi-Org: Organizations](https://trello.com/c/zr7AlfwI/316-multi-org-organizations)

**Pull request merged**